plugins/modules/remote_management/ipmi/ipmi_power.py

```diff
@@ -133,18 +210,54 @@ def main():
         )
         module.debug('ipmi instantiated - name: "%s"' % name)
 
-        current = ipmi_cmd.get_power()
-        if current['powerstate'] != state:
-            response = {'powerstate': state} if module.check_mode else ipmi_cmd.set_power(state, wait=timeout)
-            changed = True
+        changed = False
+        if not machine:
```

Comment:
```suggestion
        if machine is None:
```
Otherwise you can have `machine == []` and `state == None`.